app/services/school_groups/categorise_schools.rb

```diff
@@ -42,5 +37,35 @@ def query
         ORDER BY advice_pages.key, schools.name;
       SQL
     end
+
+    def init_hash
+      Hash.new do |results, fuel_type|
+        results[fuel_type] = Hash.new do |advice_pages, advice_page_key|
+          advice_pages[advice_page_key] = Hash.new do |benchmarks, benchmarked_as|
+            benchmarks[benchmarked_as] = []
+          end
+        end
+      end
+    end
+  end
+
+  class SchoolResult
```

Comment:
nice refactor!   maybe the `SchoolResult` class could be moved to a seperate file in the `app/services/school_groups/` directory?